scripts/convert_merged_tract_to_dpdd.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+
+"""Take LSST DESC DC2 Generic Catalog Reader accessible Object table
+(merged_tract trim cat file) and produce output files labeled with DPDD names.
+
+Output in HDF, FITS, and Parquet
+
+For each format there is one file per tract+patch.
+
+TODO: Exploring writing HDF and Parquet files in full tracts
+because those file formats allow easy writing to existing files.
+Keep FITS files in tracdt+patch.
+
+Requires
+
+pandas  # version >= 0.21
+generic-catalog-reader
+LSSTDESC/gcr-catalog
+
+and either
+pyarrow or fastparquet.
+"""
+
+import sys
+
+from astropy.table import Table
+import pandas as pd
+
+import GCRCatalogs
+
+
+def convert_all_to_dpdd(reader='dc2_coadd_run1.1p', **kwargs):
+    """Produce DPDD output files for all available tracts in GCR 'reader'."""
+    trim_config = {'filename_pattern': 'trim_merged_tract_.*\.hdf5$'}
+    cat = GCRCatalogs.load_catalog(reader, trim_config)
+
+    convert_cat_to_dpdd(cat, **kwargs)
+
+
+def convert_tract_to_dpdd(tract, reader='dc2_coadd_run1.1p', **kwargs):
+    """Produce DPDD output files for specified 'tract' and GCR 'reader'."""
+    trim_thistract_config = {
+        'filename_pattern': 'trim_merged_tract_{:04d}\.hdf5$'.format(tract)}
+    cat = GCRCatalogs.load_catalog(reader, trim_thistract_config)
+
+    convert_cat_to_dpdd(cat, **kwargs)
+
+
+def convert_cat_to_dpdd(cat, reader='dc2_coadd_run1.1p',
+                        key_prefix='object',
+                        verbose=True, **kwargs):
+    """Save DPDD columns for all tracts, patches in a input GCR catalog.
+
+    This is done in one function because we're doing this chunk-by-chunk.
+    """
+    columns = cat.list_all_quantities()
+    columns.extend(['tract', 'patch'])
+
+    # tract is an int
+    # but patch is a string (e.g., '01' for '0,1')
+    outfile_base_format = '{base}_tract_{tract:04d}_patch_{patch:s}'
+    quantities = cat.get_quantities(columns, return_iterator=True)
+    for q in quantities:
+        if 'tract' not in q or len(q) < 1:
```

Comment:
Would this ever happen?  I don't think so.

Reply:
Yes, it did.  That's why I added that check.  I got an error message about `tract` not being found.

But really I bet that the check just needs to be for `len(q) < 1`.

Reply:
And now I can't reproduce the error...

Reply:
I checked GCR source code again and still didn't think the reader will ever return an empty dictionary. 

I suggest removing this check. If an empty dictionary is returned, it's useful to get an exception because it is not expected.

Reply:
Thanks for checking.  Check removed.